src/test/java/spoon/test/ctCase/CtCaseImplTest.java

```diff
@@ -0,0 +1,58 @@
+package spoon.test.ctCase;
+
+import org.junit.jupiter.api.Test;
+import spoon.Launcher;
+import spoon.reflect.code.CtCase;
+import spoon.reflect.code.CtStatement;
+import spoon.reflect.code.CtStatementList;
+import spoon.reflect.declaration.CtClass;
+import spoon.reflect.visitor.filter.TypeFilter;
+import spoon.support.compiler.FileSystemFile;
+import spoon.support.reflect.code.CtStatementListImpl;
+import spoon.test.ctCase.testclasses.ClassWithSwitchExample;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class CtCaseImplTest {
+
+    @Test
+    void testInsertBeginWithListOfStatements() {
+        // contract: insertBegin adds a list of statements at the beginning of a case, i.e the list is added above two
+        // exiting statements, and the order of statements in the list remains the same
+
+        // arrange
+        CtClass classWithSwitchExample = getClassWithSwitchExample();
+        CtStatement firstStatementToBeInserted = createStatement(classWithSwitchExample, "firstStatement");
+        CtStatement secondStatementToBeInserted = createStatement(classWithSwitchExample, "secondStatement");
+        CtCase testCase = classWithSwitchExample.getElements(new TypeFilter<>(CtCase.class)).get(0);
```

Comment:
Always prefer writing test cases without using a test resource when it's not too complicated to do so. That makes the test cases more self-contained. In this case, it's very easy to just create the `CtCase` you want to test with a factory.

```suggestion
        Factory factory = new Launcher().getFactory();
        CtCase<?> testCase = factory.createCase();
        testCase.addStatement(factory.createCodeSnippetStatement("int preexisting = 42;").compile());

        CtStatement firstStatementToBeInserted = factory.createCodeSnippetStatement("int first = 1;").compile();
        CtStatement secondStatementToBeInserted = factory.createCodeSnippetStatement("int second = 2;").compile();
```

Reply:
Good names on the statements (e.g. `firstStatementToBeInserted`) by the way!